src/target/riscv/riscv-013.c

```diff
@@ -552,6 +553,11 @@ static bool check_dbgbase_exists(struct target *target)
 			LOG_TARGET_ERROR(target, "Reached the end of DM chain (detected %u DMs in total).", count);
 			break;
 		}
+		if (next_dm > (1U << info->abits) - 1) {
```

Comment:
[nitpick]
```suggestion
		if (next_dm >= 1U << info->abits) {
```
IMHO, it's a tiny bit simpler this way.

Reply:
Thanks, I just thought that can use `if (next_dm >> info->abits)`